source/game/system/map/MapdataCheckPath.cc

```diff
@@ -0,0 +1,32 @@
+#include "game/system/map/MapdataCheckPath.hh"
+
+namespace System {
+
+MapdataCheckPath::MapdataCheckPath(const SData *data) : m_rawData(data), m_depth(-1) {
+    u8 *unsafeData = reinterpret_cast<u8 *>(const_cast<SData *>(data));
+    EGG::RamStream stream = EGG::RamStream(unsafeData, sizeof(SData));
+    read(stream);
+    m_oneOverCount = 1.0f / m_size;
+}
+
+void MapdataCheckPath::read(EGG::Stream &stream) {
+    m_start = stream.read_u8();
+    m_size = stream.read_u8();
+    for (u8 &prev : m_prev) {
+        prev = stream.read_u8();
+    }
+
+    for (u8 &next : m_next) {
```

Comment:
```suggestion
    for (auto &prev : m_prev) {
        prev = stream.read_u8();
    }

    for (auto &next : m_next) {
```
Prefer `auto` for foreach loops.